clock.go

```diff
@@ -0,0 +1,158 @@
+/*
+management-interface - Web based management of Raspberry Pis over WiFi
+Copyright (C) 2018, The Cacophony Project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package managementinterface
+
+import (
+	"log"
+	"net/http"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+)
+
+// timeSettings is a struct which hold the time settings.
+type timeSettings struct {
+	RTCTime    time.Time
+	SystemTime time.Time
+}
+
+func getTimes() (*timeSettings, error) {
+	if runtime.GOOS != "windows" {
+		// 'Nix.  Run hwclock command to get the times we want.
+		ts := &timeSettings{SystemTime: time.Now()}
+		out, err := exec.Command("/sbin/hwclock", "-r").Output()
+		if err != nil {
+			log.Printf(err.Error())
```

Comment:
instead of both logging and returning the error, it's better practice to just return it and let a higher layer take care of logging/handling it.